frontend/src/App.test.tsx

```diff
@@ -475,3 +480,99 @@ describe("App.handleNewReport", () => {
     expect(SessionInfo.isSet()).toBe(true)
   })
 })
+
+// Using this to test the functionality provided through streamlit.experimental_set_query_params.
+// Please see https://github.com/streamlit/streamlit/issues/2887 for more context on this.
+describe("App.handlePageInfoChanged", () => {
+  beforeEach(() => {
+    // Reset the value of document.location.pathname.
+    window.history.pushState({}, "", "/")
+  })
+  it("does not override the pathname when setting query params", () => {
+    const wrapper = shallow(<App {...getProps()} />)
+    const app = wrapper.instance()
```

Comment:
I'd recommend shortening the tests a bit by declaring `wrapper` and `app` variables at the top of the `describe` block and setting them inside the `beforeEach` (note we have linter rules to enforce that `let` is used instead of `var`). Other `describe` blocks repeatedly do this in the tests since they need to change `props` from time to time, but since we don't need to do that here we can make tests a bit less redundant